src/libs/actions/OnyxUpdateManager/utils/index.ts

```diff
@@ -33,37 +33,54 @@ function detectGapsAndSplit(updates: DeferredUpdatesDictionary, clientLastUpdate
     for (const [index, update] of updateValues.entries()) {
         const isFirst = index === 0;
 
+        const lastUpdateID = Number(update.lastUpdateID);
+        const previousUpdateID = Number(update.previousUpdateID);
+
+        // If an update is older than the last update that was applied to the client, we can skip it
+        // This should already be handled before, but we'll keep this as a safeguard.
+        if (lastUpdateID <= lastUpdateIDFromClient) {
+            // eslint-disable-next-line no-continue
+            continue;
+        }
+
         // If any update's previousUpdateID doesn't match the lastUpdateID from the previous update, the deferred updates aren't chained and there's a gap.
         // For the first update, we need to check that the previousUpdateID of the fetched update is the same as the lastUpdateIDAppliedToClient.
         // For any other updates, we need to check if the previousUpdateID of the current update is found in the deferred updates.
         // If an update is chained, we can add it to the applicable updates.
-        const isChained = isFirst ? update.previousUpdateID === lastUpdateIDFromClient : !!updates[Number(update.previousUpdateID)];
+        const isChained = isFirst ? previousUpdateID === lastUpdateIDFromClient : !!updates[Number(update.previousUpdateID)];
         if (isChained) {
             // If a gap exists already, we will not add any more updates to the applicable updates.
             // Instead, once there are two chained updates again, we can set "firstUpdateAfterGaps" to the first update after the current gap.
             if (gapExists) {
                 // If "firstUpdateAfterGaps" hasn't been set yet and there was a gap, we need to set it to the first update after all gaps.
                 if (!firstUpdateAfterGaps) {
-                    firstUpdateAfterGaps = Number(update.previousUpdateID);
+                    firstUpdateAfterGaps = previousUpdateID;
                 }
             } else {
                 // If no gap exists yet, we can add the update to the applicable updates
-                applicableUpdates[Number(update.lastUpdateID)] = update;
+                applicableUpdates[lastUpdateID] = update;
             }
         } else {
+            // If the previousUpdateID of the update after the gap is older than the lastUpdateIDFromClient,
+            // we don't want to detect a missing update, since this would cause a recursion loop in "validateAndApplyDeferredUpdates".
+            if (previousUpdateID <= lastUpdateIDFromClient) {
```

Comment:
Can we just raise this condition up next to the other `if (lastUpdateID <= lastUpdateIDFromClient)` check? I think it's less confusing to follow there

Reply:
This shouldn't break the upper `if` block, because there might be updates, that have an "outdated" previous update, but are still valid to be applied.

I used the chance to improve the code a bit and further improve the comments, so this might be more clear now.